### PR TITLE
Add RelationshipTest for Relationship testing

### LINKED ISSUE #93: Add test cases for Relationship
Currently there are no test cases for the Relationship class.

Let's add some test cases which test the following: 
- Check for NullPointerException for a null Relationship
- Check if a Relationship created is valid based on the provided `VALIDATION_REGEX`
- Checks if the `equals` method of Relationship works as expected (for different types, null etc.)

### PR DESCRIPTION
**Pull request by qilstiano**
Currently, the Relationship class lacks comprehensive unit tests to ensure its functionality and validation logic works as intended.

Without proper test coverage, we risk introducing bugs or regressions when modifying Relationship in the future. Comprehensive tests help maintain code quality and catch potential issues early.

Let's add a RelationshipTest class to test the Relationship class. In particular, we test for:
- Constructor behaviour with null and invalid inputs
- Validation of various relationship strings to ensure that `VALIDATION_REGEX` works as intended
- Equality comparisons using the `equals` method

The test class follows the conventions previously laid out by other Test classes and covers various basic scenarios to ensure robust validation of Relationship. It includes various positive and negative test cases, helping to verify the correctness of the implementation of the Relationship class.

For future commits: Let's continue to add more edge cases in order to truly gauge the functionality of Relationship! (e.g. non ASCII characters, symbol only checks etc.)

Fixes #93

### CODE REVIEW

File: src/test/java/seedu/address/model/person/RelationshipTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Relationship(null));
+    }
+
+    @Test
+    public void constructor_invalidRelationship_throwsIllegalArgumentException() {
+        String invalidRelationship = "";
+        assertThrows(IllegalArgumentException.class, () -> new Relationship(invalidRelationship));
+    }
```
[the remark this answers is not in the document]
Comment (qilstiano) [reply src/test/java/seedu/address/model/person/RelationshipTest.java:20]: image

**Review by nictjh — APPROVED** [review]
Nice job testing the relationship class in general. Looks good and comprehensive for now!

File: src/test/java/seedu/address/model/person/RelationshipTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (qilstiano) [reply src/test/java/seedu/address/model/person/RelationshipTest.java:20]: image

File: src/test/java/seedu/address/model/person/RelationshipTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Test
+    public void isValidRelationship() {
```
[the remark this answers is not in the document]
Comment (qilstiano) [reply src/test/java/seedu/address/model/person/RelationshipTest.java:23]: image

Thanks for the review bro!

File: src/test/java/seedu/address/model/person/RelationshipTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        assertFalse(Relationship.isValidRelationship("")); // empty string
+        assertFalse(Relationship.isValidRelationship("^")); // only non-alphanumeric characters
+        assertFalse(Relationship.isValidRelationship("mother*")); // contains non-alphanumeric characters
+
+        // valid relationship
+        assertTrue(Relationship.isValidRelationship("mother")); // alphabets only
+        assertTrue(Relationship.isValidRelationship("12345")); // numbers only
+        assertTrue(Relationship.isValidRelationship("father 2nd")); // alphanumeric characters
+        assertTrue(Relationship.isValidRelationship("Grand Father")); // with capital letters
+        assertTrue(Relationship.isValidRelationship("Great Grand Mother")); // long relationships
```
[the remark this answers is not in the document]
Comment (qilstiano) [reply src/test/java/seedu/address/model/person/RelationshipTest.java:37]: Appreciate the comprehensive feedback that you and Yashvan have provided! Thanks for reviewing and merging guys 🔥 

image
